internal/repository/file.go

```diff
@@ -171,6 +173,11 @@ func (r *FileRepository) Parent(u fyne.URI) (fyne.URI, error) {
 	// trim the scheme
 	s = strings.TrimPrefix(s, fileSchemePrefix)
 
+	// Only for Windows: If the path is a drive root, no parent is possible
+	if runtime.GOOS == "windows" && regexp.MustCompile(`(?i)^/[a-z]:$`).MatchString(s) {
```

Comment:
Compiling the regexp each time is going to be incredibly slow. It is better to cache this in some way. The initial thought would be to use a global variable but it seems unnecessary to do that for all operating systems.

Reply:
I suspect there are not thousands of these calls per second in 99.99% of all applications, but I can for example move it to a local "global" variable and compile it only once there. Or to a utility method and put the global there. The downside is: globals are not nice.
The alternative could be a utility struct like:
```go
var windowsDrivePattern = regexp.MustCompile(`(?i)^/[a-z]:$`)
type uriUtility struct {
}

func (f *uriUtility) MatchesDrivePattern(path string) bool {
	return windowsDrivePattern .MatchString(path)
}
```

Reply:
Personally I have no problem with a `var windowsDrivePattenRegex = regexp.MustCompile(...)` or similar name added to the top of this file, and then just matching directly against it here (ie no need for `uriUtility`). @Jacalz do you have a different opinion?

Reply:
I added the variables now.